torch/csrc/distributed/c10d/ProcessGroupGloo.cpp

```diff
@@ -465,9 +469,30 @@ ProcessGroupGloo::AsyncWork::AsyncWork(
     std::vector<std::vector<at::Tensor>> outputTensors,
     const char* profilingTitle,
     const c10::optional<std::vector<at::Tensor>>& inputTensors)
-    : ProcessGroup::Work(-1, OpType::UNKNOWN, profilingTitle, inputTensors),
```

Comment:
In one sense it's a shame to lose the generic ProcessGroup event since it will be something present in the trace regardless of implementation. Does it still make sense to keep it, and if so is it possible without causing overlapping regions?

Reply:
Yep. The current generic ProcessGroup event implementation is for sync work and it do not take into account the async scenario. Also, it's difficult for generic ProcessGroup profiling the sync work and async work in the same way and we should process the two scenarios separately.